code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm

```diff
@@ -354,13 +354,25 @@
 	)
 	heal_range = HIVELORD_HEAL_RANGE
 	target_flags = ABILITY_MOB_TARGET
+	var/hivemind_heal = FALSE
 
 /datum/action/ability/activable/xeno/psychic_cure/queen_give_heal/use_ability(atom/target)
 	if(owner.do_actions)
 		return FALSE
-	if(!do_after(owner, 1 SECONDS, NONE, target, BUSY_ICON_FRIENDLY, BUSY_ICON_MEDICAL))
-		return FALSE
-	target.visible_message(span_xenowarning("\the [owner] vomits acid over [target], mending their wounds!"))
+	if(!hivemind_heal)
+		if(!do_after(owner, 1 SECONDS, NONE, target, BUSY_ICON_FRIENDLY, BUSY_ICON_MEDICAL))
+			return FALSE
+		target.visible_message(span_xenowarning("\the [owner] vomits acid over [target], mending their wounds!"))
+	else
+		if(!do_after(owner, 1 SECONDS, IGNORE_TARGET_LOC_CHANGE, target, BUSY_ICON_FRIENDLY, BUSY_ICON_MEDICAL))
+			return FALSE
+		if(!can_use_ability(target, TRUE))
+			return FALSE
```

Comment:
you are still copy pasting stuff twice.
Just have 1 do_after, they're both the same except the loc change flag, so just make that a ternary based on your new var.
the can_use_ability proc should be checked regardless as well.

Then just make the message be behind the an if.